### PR TITLE
DOCSP-41143: Compound Indexes

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41143>
Staging - <https://preview-mongodbmcmorisi.gatsbyjs.io/kotlin-sync/DOCSP-41143-compound-indexes/indexes/compound-index/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/includes/indexes/indexes.kt [hunk 1]
```diff
@@ -51,4 +52,21 @@ fun main() {
...
     // end-index-single-query
+
+    // start-index-compound
+    collection.createIndex(Indexes.ascending(Movie::type.name, Movie::genres.name))
+    // end-index-compound
+
+    // start-index-compound-query
+    val filter = and(
+        eq(Movie::type.name, "movie"),
+        `in`(Movie::genres.name, "Drama")
```
Comment (mcmorisi) [inline source/includes/indexes/indexes.kt:63]: This is a Kotlin (and probably Java) quirk – `in` is a language keyword AND a `Filters` method. To use the method as such you need to surround it with backticks.

File: source/indexes/compound-index.txt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+.. literalinclude:: /includes/indexes/indexes.kt
+   :start-after: start-index-compound
+   :end-before: end-index-compound
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+The following is an example of a query that uses the index created in
+the preceding code example:
```
Comment (stephmarie17) [inline source/indexes/compound-index.txt:61]: [s] Maybe this introduction can be reworded to improve clarity? It can be hard to parse "following example" vs "preceding example" all in one sentence.

File: source/indexes/compound-index.txt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Overview
+--------
+
+:manual:`Compound indexes </core/index-compound/>` hold references to multiple
+fields within a collection's documents, improving query and sort performance.
+
+When creating a compound index, you must specify the following:
+
+- The fields on which to create the index
+- The sort order for each field (ascending or descending)
```
Comment (stephmarie17) [inline source/indexes/compound-index.txt:29]: [q] Is there a default sort order if the sort order isn't specified or is specifying always required?

File: source/indexes/compound-index.txt [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+      Movie(id=573a1392f29313caabcd9ca6, title=Scarface, type=movie, genres=[Action, Crime, Drama], ... )
+      Movie(id=573a1392f29313caabcdb258, title=The Hurricane, type=movie, genres=[Action, Drama, Romance], ...)
+      Movie(id=573a1391f29313caabcd820b, title=Beau Geste, type=movie, genres=[Action, Adventure, Drama], ...)
+      ...
+
+Additional Information
+----------------------
+
+To learn more about compound indexes, see :manual:`Compound Indexes </core/index-compound>`
+in the {+mdb-server+} manual.
```
Comment (stephmarie17) [inline source/indexes/compound-index.txt:85]: [s] Would linking to this resource be helpful for the reader: https://www.mongodb.com/docs/manual/tutorial/equality-sort-range-rule/#std-label-esr-indexing-rule

**Review by stephmarie17 — APPROVED** [review]
LGTM 🚀 left a few small comments, but once the test gets sorted out should be gtg

File: source/indexes/compound-index.txt (hunk 3 above)
Comment (mcmorisi) [reply source/indexes/compound-index.txt:29]: It's required – [source](https://www.mongodb.com/docs/manual/core/indexes/index-types/index-compound/)

**Review by jyemin — APPROVED** [review]
